src/librustdoc/html/render/cache.rs

```diff
@@ -373,18 +373,19 @@ crate fn get_all_types<'tcx>(
     tcx: TyCtxt<'tcx>,
 ) -> (Vec<TypeWithKind>, Vec<TypeWithKind>) {
     let mut all_types = Vec::new();
+    // This variable is shared between iterations to reduce memory usage (since the `Vec` won't be
+    // reallocated every time).
+    let mut args = Vec::new();
+
     for arg in decl.inputs.values.iter() {
         if arg.type_.is_self_type() {
             continue;
         }
-        // FIXME: performance wise, it'd be much better to move `args` declaration outside of the
-        // loop and replace this line with `args.clear()`.
-        let mut args = Vec::new();
```

Comment:
You forgot to add `args.clear()`. `get_real_types` only appends, it doesn't replace.

Reply:
I added a comment to explain this just below: `append` empties `args` (as you can see from [append documentation](https://doc.rust-lang.org/nightly/std/vec/struct.Vec.html#method.append)). Unless you were talking about something else?

Reply:
Ah ok. That's not obvious, thanks for adding the comment.

Reply:
No it's not, which is why @camelid suggested that if no improvement was seen from perf run, we would not merge this and simply remove the FIXME comments to keep the code easier to read.